nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/DebugTask.java

```diff
@@ -73,28 +73,42 @@ public String debug() throws MojoExecutionException {
 
         assert (!this.failingConfigurations.isEmpty());
 
-        Configuration failingOne = this.debugWithConfigurations(this.failingConfigurations);
+        List<Configuration> debuggedOnes = this.debugWithConfigurations(this.failingConfigurations);
 
         Logger.getGlobal().log(Level.SEVERE, "limits : " + limits.getLeft() + "  " + limits.getRight());
 
-        if (failingOne != null) {
-            return failingOne.toArgLine() + "\nDEBUG RESULTS FOR " + failingOne.testName + " AT: "
-                + failingOne.getDebugPath();
+        if (debuggedOnes.size() > 0) {
+            return makeResultString(debuggedOnes);
         }
 
         // The seeds that failed with the full test-suite no longer fail
         // Searching for different seeds
+        Logger.getGlobal().log(Level.FINE, "TRYING NEW SEEDS");
         List<Configuration> retryWOtherSeeds = this.createNewSeedsToRetry();
-        failingOne = this.debugWithConfigurations(retryWOtherSeeds);
+        debuggedOnes = this.debugWithConfigurations(retryWOtherSeeds);
 
-        if (failingOne != null) {
-            return failingOne.toArgLine() + "\nDEBUG RESULTS FOR " + failingOne.testName + " AT: "
-                + failingOne.getDebugPath();
+        if (debuggedOnes.size() > 0) {
+            return makeResultString(debuggedOnes);
         }
 
         return "cannot reproduce. may be flaky due to other causes";
     }
 
+    private String makeResultString(List<Configuration> debuggedOnes) {
+        StringBuilder sb = new StringBuilder();
+        for (Configuration config : debuggedOnes) {
+            sb.append(config.toArgLine());
+            sb.append("\nDEBUG RESULTS FOR ");
+            sb.append(config.testName);
+            sb.append(" AND SEED: ");
+            sb.append(config.seed);
+            sb.append(" AT: ");
+            sb.append(config.getDebugPath());
+            sb.append("\n");
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid appending characters as strings in StringBuffer.append.](https://www.codacy.com/app/gyori/NonDex/file/2760301675/issues/source?bid=3322087&fileBranchId=3360609#l107)